gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/iapi/db/ConsistencyChecker.java

```diff
@@ -504,170 +530,267 @@ public static boolean checkTable(String schemaName, String tableName)
+//        null, ScanController.GT, null);
+//    DataValueDescriptor[] indexRowArray;
+//    while (indexScan.next()) {
+//      indexRowArray = indexRow.getRowArray();
+//      indexScan.fetch(indexRow);
+//      key = new CompositeRegionKey(getNewArray(indexRowArray, baseColumns,
+//          false));
+//      if (!tableIndexKeys.contains(key)) {
+//        throw StandardException.newException(
+//            SQLState.LANG_INCONSISTENT_ROW_LOCATION, td.getQualifiedName(),
+//            indexCD.getConglomerateName(), key.toString(), indexRow.toString());
+//      }
+//      ++numIndexEntries;
+//    }
+//    if (numIndexEntries != tableIndexKeys.size()) {
+//      throw StandardException.newException(
+//          SQLState.LANG_INDEX_ROW_COUNT_MISMATCH,
+//          indexCD.getConglomerateName(), td.getSchemaName(), td.getName(),
+//          numIndexEntries, tableIndexKeys.size());
+//    }
+//    if (indexScan != null) {
+//      indexScan.close();
+//      indexScan = null;
+//    }
+//    scan.close();
+//    scan = null;
+//  }
+
+	static void checkGlobalHashIndex(TransactionController tc, TableDescriptor td,
+			ConglomerateController baseCC, ConglomerateDescriptor heapCD,
+			ConglomerateDescriptor indexCD) throws StandardException {
+		LocalRegion globalIndexRegion = (LocalRegion)Misc.getRegionForTable(
+				td.getSchemaName() + "." + indexCD.getDescriptorName(), true);
+
+		ExecRow baseRow = td.getEmptyExecRow();
+		//DataValueDescriptor[] baseObjectArray = baseRow.getRowArrayClone();
+
+		IndexRowGenerator irg = indexCD.getIndexDescriptor();
+		int[] baseColumnPositions = irg.baseColumnPositions();
+		int baseColumns = baseColumnPositions.length;
+		final DataValueDescriptor[] searchCondition =
+				new DataValueDescriptor[baseColumnPositions.length];
+		ExecIndexRow indexRow = irg.getIndexRowTemplate();
+
+		ScanController scan = tc.openScan(
+				heapCD.getConglomerateNumber(),
+				false, // hold
+//				GfxdConstants.SCAN_OPENMODE_FOR_GLOBALINDEX // get proper RowLocation
+//						| GfxdConstants.SCAN_OPENMODE_FOR_FULLSCAN, // open a full table scan
+				0,
+				TransactionController.MODE_TABLE,
+				TransactionController.ISOLATION_SERIALIZABLE, RowUtil.EMPTY_ROW_BITSET,
+				null, // startKeyValue
+				0, // not used with null start posn.
+				null, // qualifier
+				null, // stopKeyValue
+				0, null); // not used with null stop posn.
+		irg.getIndexRow(baseRow, scan.newRowLocationTemplate(), indexRow, null);
+
+		final HashSet<Object> tableIndexKeys =
+				new HashSet();
+		ScanController indexScan = null;
+		Object key;
+		int numRows = 0;
+		while (scan.next()) {
+			numRows++;
+			scan.fetch(baseRow);
+			for (int i = 0; i < baseColumns; ++i) {
+				searchCondition[i] = baseRow.getColumn(baseColumnPositions[i]);
+			}
+			if (searchCondition.length == 1) {
+				key = searchCondition[0].getClone();
+			} else {
+				key = new CompositeRegionKey(getNewArray(searchCondition,
+						baseColumns, true));
```

Comment:
Since we are using a fixed batch of size 1000 then we can just have 1000 empty object created upfront and reuse the same again and again. Just using some setter methods.